### PR TITLE
1.7.10 version 0.29.4

### PR DESCRIPTION
**Pull request by H-QueerCoded**
It fixes the following bugs:
* Jetpacks always have fall damage in multiplayer
* Pyrophobic coatings crash the game when walking on lava in multiplayer

### CODE REVIEW

File: src/main/java/flaxbeard/steamcraft/block/BlockPipe.java [hunk 1]
```diff
@@ -353,8 +353,7 @@ public MovingObjectPosition collisionRayTrace(World world, int x, int y, int z,
             }
             List<IndexedCuboid6> cuboids = new LinkedList();
             ((TileEntitySteamPipe) tile).addTraceableCuboids(cuboids);
-            MovingObjectPosition mop = this.rayTracer.rayTraceCuboids(new Vector3(start), new Vector3(end), cuboids, new BlockCoord(x, y, z), this);
-            return mop;
+           return  this.rayTracer.rayTraceCuboids(new Vector3(start), new Vector3(end), cuboids, new BlockCoord(x, y, z), this);
```
Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/block/BlockPipe.java]: Irrelevant change with wrong spacing

File: src/main/java/flaxbeard/steamcraft/block/BlockSteamcraftOre.java [hunk 2]
```diff
@@ -54,18 +54,21 @@ public IIcon getIcon(int side, int meta) {
...
         	}
+        	break;
         case 1:	//Nether
         	switch(meta){
         	case 0: return this.icon[5];
         	case 1: return this.icon[6];
         	case 2: return this.icon[2];
         	}
+        	break;
         default: //Same as overworld
```
Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/block/BlockSteamcraftOre.java]: Irrelevant change

File: src/main/java/flaxbeard/steamcraft/integration/ic2/IC2RecipeInput.java [hunk 3]
```diff
@@ -35,7 +35,6 @@ public int getAmount() {
 
     @Override
     public List<ItemStack> getInputs() {
-    	List<ItemStack> list = Arrays.asList(input);
-        return list;
+    	return Arrays.asList(input);
```
Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/integration/ic2/IC2RecipeInput.java]: Irrelevant change

File: src/main/java/flaxbeard/steamcraft/integration/tinkers/ModiferSteam.java [hunk 4]
```diff
@@ -17,17 +17,20 @@ public ModiferSteam(ItemStack[] tank) {
...
     protected boolean canModify(ItemStack tool, ItemStack[] input) {
-        NBTTagCompound tags = tool.getTagCompound().getCompoundTag("InfiTool");
-
-        if (tags.getInteger("Modifiers") > 0 && !tags.getBoolean(key)) {
-            return false;
-        }
-        
-        if (Arrays.asList(((ToolCore) tool.getItem()).getTraits()).contains("ammo")) {
-            return false;
+        if(tool != null && tool.getItem() != null && tool.getItem() instanceof ToolCore) {
```
Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/integration/tinkers/ModiferSteam.java]: Space in between `if` and `(`

File: src/main/java/flaxbeard/steamcraft/tile/TileEntityVacuum.java [hunk 5]
```diff
@@ -73,11 +73,10 @@ public static boolean isLyingInCone(float[] x, float[] t, float[] b, float apert
         // X is contained in cone only if projection of apexToXVect to axis
         // is shorter than axis.
         // We'll use dotProd() to figure projection length.
-        boolean isUnderRoundCap = dotProd(apexToXVect, axisVect)
+        return  dotProd(apexToXVect, axisVect)
```
Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/tile/TileEntityVacuum.java]: Irrelevant change

File: src/main/java/flaxbeard/steamcraft/tile/TileEntitySteamGauge.java [hunk 6]
```diff
@@ -50,9 +50,8 @@ public float getPressure() {
     }
 
     public int getComparatorOutput() {
-        int out = (int) (15 * (100 * ((double) getPressure() * 0.01D)));
+        return (int) (15 * (100 * ((double) getPressure() * 0.01D)));
```
Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/tile/TileEntitySteamGauge.java]: Irrelevant change

File: src/main/java/flaxbeard/steamcraft/tile/TileEntityCrucible.java [hunk 7]
```diff
@@ -276,8 +276,7 @@ public void setTipping() {
     }
 
     public int getComparatorOutput() {
-        int out = (int) ((double) 15 * (((double) getFill() / 90D)));
```
Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/tile/TileEntityCrucible.java]: Irrelevant change

File: src/main/java/flaxbeard/steamcraft/integration/tinkers/ModiferSteam.java [hunk 8]
```diff
@@ -17,17 +17,20 @@ public ModiferSteam(ItemStack[] tank) {
 
     @Override
     protected boolean canModify(ItemStack tool, ItemStack[] input) {
```
Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/integration/tinkers/ModiferSteam.java]: Irrelevant change

**Review by elifoster — CHANGES_REQUESTED** [review]
Please remove the irrelevant changes to the codebase, then I will test and re-review this PR. Thank you for your contributions!

**Comment by H-QueerCoded** [comment]
All those changes came from merging the source from the latest release with the branch, if you want I can remove them, but I just wanted to note that they are a result of discrepancies between the 1.7 branch and the 0.29.3 release, not changes I made.

**Comment by elifoster** [comment]
Oh, okay. Hold off on that then, I'm going to try to figure out why those commits are not in the 1.7 branch.

**Comment by elifoster** [comment]
Okay, the 1.7 branch has been updated to include the missing commits. This was so long ago, I don't remember what happened but somehow all the commits from 0.29.3 disappeared and were only accessible through the 0.29.3 tag. Anyway, if you update your 1.7 branch and fix the merge conflicts it should be fine.

**Comment by H-QueerCoded** [comment]
I got it updated

**Comment by elifoster** [comment]
Cool, will find some time to test it out soon.

**Pull request closed without merging**